### PR TITLE
BUG Fix for Issue #12565 - font size on secondary_y

### LINKED ISSUE #12565: BUG?: secondary_y is unaltered by fontsize parameter
Hi,

I have found that fontsize is ignored on the secondary_y axis in DataFrame.plot in version 17.1.  Is this by intention?  (Apologies, if this is the wrong place to ask this or if it has already been asked as I was unable to find it.)

Thanks!

Example:

``` python
>>> import pandas as pd, numpy as np, matplotlib.pyplot as plt
>>> x = pd.DataFrame(np.random.randn(100,2), columns=list('AB')).assign(C = lambda df: df.B.cumsum())
>>> x.plot(secondary_y='C', fontsize=20).legend(fontsize=16)
<matplotlib.legend.Legend object at 0x05F0B990>
>>> plt.show() # Righthand y axis remains small
```

Here are my package versions:

``` python
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.9.final.0
python-bits: 32
OS: Windows
OS-release: 8
machine: AMD64
processor: AMD64 Family 21 Model 16 Stepping 1, AuthenticAMD
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.17.1
nose: None
pip: 1.5.6
setuptools: 7.0
Cython: None
numpy: 1.9.3
scipy: 0.16.0
statsmodels: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.0
openpyxl: None
xlrd: 0.9.4
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
Jinja2: None
```

### PR DESCRIPTION
**Pull request by Morgan243**
There may be other plotting methods that need testing.

 - [ ] closes #12565
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master -u -- "*.py" | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/plotting/test_frame.py [hunk 1]
```diff
@@ -2733,6 +2733,22 @@ def test_rcParams_bar_colors(self):
                 barplot = pd.DataFrame([[1, 2, 3]]).plot(kind="bar")
         assert color_tuples == [c.get_facecolor() for c in barplot.patches]
 
+    @pytest.mark.parametrize('method', ['line', 'barh', 'bar'])
+    def test_secondary_axis_font_size(self, method):
+        df = (pd.DataFrame(np.random.randn(15, 2),
```
Comment (TomAugspurger) [inline pandas/tests/plotting/test_frame.py]: Can add comment with the github issue number here like `# GH: 12565`

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -166,7 +166,7 @@ I/O
 
 Plotting
 ^^^^^^^^
-
+- Bug in plotting methods using ``secondary_y`` and ``fontsize`` not setting secondary axes font size (:issue:`12565`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: "axis" not "axes" ?

Comment (Morgan243) [inline doc/source/whatsnew/v0.21.0.txt]: I agree, I think 'axis' is right

**Review by gfyoung — APPROVED** [review]

**Comment by gfyoung** [comment]
@Morgan243 : Ping when tests pass, and we will merge this!

**Comment by Morgan243** [comment]
@gfyoung tests passed!

**Comment by gfyoung** [comment]
Thanks @Morgan243 !

**Pull request merged**